controller_manager:
  ros__parameters:
    update_rate: 50
    use_sim_time: true   # <-- set true for sim

    joint_state_broadcaster.type: joint_state_broadcaster/JointStateBroadcaster
    platform_velocity_controller.type: diff_drive_controller/DiffDriveController
    xarm6_traj_controller.type: joint_trajectory_controller/JointTrajectoryController

platform_velocity_controller:
  ros__parameters:
    use_sim_time: true   # <-- set true for sim
    left_wheel_names:  ["a200_front_left_wheel_joint", "a200_rear_left_wheel_joint"]
    right_wheel_names: ["a200_front_right_wheel_joint", "a200_rear_right_wheel_joint"]

    wheel_separation: 0.555
    wheel_radius: 0.1651
    wheels_per_side: 2         # youâ€™ve got two per side; set 2

    publish_rate: 50.0
    odom_frame_id: odom
    base_frame_id: a200_base_footprint
    enable_odom_tf: true           # you can set true later if you want TF

    cmd_vel_timeout: 0.5
    use_stamped_vel: false

    linear.x.has_velocity_limits: true
    linear.x.max_velocity: 1.0
    linear.x.has_acceleration_limits: true
    linear.x.max_acceleration: 1.0

    angular.z.has_velocity_limits: true
    angular.z.max_velocity: 1.0
    angular.z.has_acceleration_limits: true
    angular.z.max_acceleration: 1.0

xarm6_traj_controller:
  ros__parameters:
    command_interfaces:
      - position
      - velocity
    state_interfaces:
      - position
      - velocity
    joints:
      - xarm6_joint1
      - xarm6_joint2
      - xarm6_joint3
      - xarm6_joint4
      - xarm6_joint5
      - xarm6_joint6
    constraints:
      goal_time: 0.5
      stopped_velocity_tolerance: 0.0
      xarm6_joint1: {trajectory: 1.0, goal: 0.01}
      xarm6_joint2: {trajectory: 1.0, goal: 0.01}
      xarm6_joint3: {trajectory: 1.0, goal: 0.01}
      xarm6_joint4: {trajectory: 1.0, goal: 0.01}
      xarm6_joint5: {trajectory: 1.0, goal: 0.01}
      xarm6_joint6: {trajectory: 1.0, goal: 0.01}
    interface_name: position
    state_publish_rate: 25.0
    action_monitor_rate: 10.0